File: backend/nutrition/domain.py
"""
Domain vocabulary — single source of truth for API validation.

These atoms MUST stay in sync with prolog/kb.pl.  Drift tests in tests.py
assert that every condition used by suitable_for/2 is listed here (and
vice-versa for the public recommendable set).
"""

# Conditions accepted by POST /api/recommend/condition/
# and produced by condition_for_deficiency/2 in the KB.
CONDITIONS = (
    "healthy",
    "hypertension",
    "type2_diabetes",
    "anaemia",
    "vitA_deficiency",
    "rickets",
    "unknown",
)

# Symptoms accepted by POST /api/recommend/symptoms/
SYMPTOMS = (
    "fatigue",
    "pale_skin",
    "night_blindness",
    "dry_skin",
    "frequent_infections",
    "bone_pain",
    "muscle_weakness",
    "rickets",
    "mouth_sores",
    "muscle_cramps",
)

# Food groups present in food/7 facts (for validation of path params).
FOOD_GROUPS = (
    "grains",
    "legumes_grains",
    "vegetables",
    "legumes",
    "tubers",
    "dairy",
    "fish",
    "meat",
    "protein",
    "nuts",
    "fruits",
)

# Activity levels for UserProfile (mirrored in the model choices).
ACTIVITY_LEVELS = (
    "sedentary",
    "light",
    "moderate",
    "active",
    "very_active",
)

# Strict Prolog atom pattern: lowercase letter then alphanumerics/underscores.
# Used to reject injection attempts before building query strings.
PROLOG_ATOM_RE = r"^[a-z][a-zA-Z0-9_]*$"

File: frontend/src/constants/domain.js
/**
 * Domain vocabulary mirrored from backend/nutrition/domain.py.
 * Keep in sync when adding conditions or symptoms.
 */

export const CONDITIONS = [
  { value: "healthy", label: "Healthy (general)" },
  { value: "hypertension", label: "Hypertension" },
  { value: "type2_diabetes", label: "Type 2 Diabetes" },
  { value: "anaemia", label: "Anaemia" },
  { value: "vitA_deficiency", label: "Vitamin A Deficiency" },
  { value: "rickets", label: "Rickets / Vitamin D Deficiency" },
];

export const SYMPTOMS = [
  { value: "fatigue", label: "Fatigue / tiredness" },
  { value: "pale_skin", label: "Pale skin" },
  { value: "night_blindness", label: "Night blindness" },
  { value: "dry_skin", label: "Dry skin" },
  { value: "frequent_infections", label: "Frequent infections" },
  { value: "bone_pain", label: "Bone pain" },
  { value: "muscle_weakness", label: "Muscle weakness" },
  { value: "rickets", label: "Rickets (bowing of legs)" },
  { value: "mouth_sores", label: "Mouth sores" },
  { value: "muscle_cramps", label: "Muscle cramps" },
];

export const ACTIVITY_OPTIONS = [
  { value: "sedentary", label: "Sedentary" },
  { value: "light", label: "Lightly Active" },
  { value: "moderate", label: "Moderately Active" },
  { value: "active", label: "Active" },
  { value: "very_active", label: "Very Active" },
];
